### ISSUE TITLE
[Question] Exception RuntimeError: Link object failed when using super mode

### ISSUE THREAD

**Issue by hongphu8790** [dashingsoft/pyarmor#823]
I have a problem when obfuscating using super mode, the exception:

RuntimeError: Link object failed

My environment:
```
Linux aarch64 
PyArmor Version 7.6.1
Python 3.9.13

Purchased pyarmor license
```

This is script:

foo.py
```
class Case1:
    def test1(self):
        print("hello")
```

This is command:
```
PYARMOR_PLATFORM=linux.aarch64.3 pyarmor -d obfuscate --advanced 5 --platform linux.aarch64 --exact foo.py
```

Verbose exception:
```
~# PYARMOR_PLATFORM=linux.aarch64.3 pyarmor -d obfuscate --advanced 5 --platform linux.aarch64 --exact foo.py
DEBUG    PyArmor installation path: /root/.pyenv/versions/3.9.13/lib/python3.9/site-packages/pyarmor
DEBUG    PyArmor home path: /root/.pyarmor
DEBUG    Native platform is linux.aarch64
INFO     PYARMOR_PLATFORM is linux.aarch64.3
DEBUG    Build platform is linux.aarch64.3
DEBUG    Search dynamic library in the path: /root/.pyenv/versions/3.9.13/lib/python3.9/site-packages/pyarmor/platforms
DEBUG    Search dynamic library in the path: /root/.pyarmor/platforms
DEBUG    Loaded dynamic library: /root/.pyarmor/platforms/linux/aarch64/3/_pytransform.so
DEBUG    The version of core library is (51, 5, 3)
INFO     PyArmor Version 7.6.1
INFO     Python 3.9.13
INFO     Target platforms: ['linux.aarch64']
DEBUG    Load platform list from /root/.pyarmor/platforms/index.json
INFO     Update target platforms to: ['linux.aarch64.11.py39']
INFO     Source path is "/root/.phu/gen"
INFO     Entry scripts are ['foo.py']
INFO     Use cached capsule /root/.pyarmor/.pyarmor_capsule.zip
INFO     Search scripts mode: Exact
INFO     Save obfuscated scripts to "dist"
INFO     Read product key from capsule
INFO     Obfuscate module mode is 2
INFO     Obfuscate code mode is 1
INFO     Obfuscate string value is False
INFO     Wrap mode is 1
INFO     Restrict mode is 1
INFO     Advanced value is 2
INFO     Super mode is True
INFO     Super plus mode is enabled
INFO     Generating super runtime library to "dist"
DEBUG    Load platform list from /root/.pyarmor/platforms/index.json
DEBUG    Target dynamic library: ['/root/.pyarmor/platforms/linux/aarch64/11/py39/pytransform.cpython-39-aarch64-linux-gnu.so']
INFO     Extract pytransform.key
INFO     Read default license from capsule
INFO     Copying /root/.pyarmor/platforms/linux/aarch64/11/py39/pytransform.cpython-39-aarch64-linux-gnu.so
INFO     Patch extension dist/pytransform.so
DEBUG    Patching dist/pytransform.so
DEBUG    Found pattern at 9e90c
DEBUG    Patch 261 bytes from 9e92c
DEBUG    Patch 270 bytes from 9ea31
DEBUG    Patch 412 bytes from 9eb3f
INFO     Patch library file OK
INFO     Generate runtime files OK
INFO     Use protection template: /root/.pyenv/versions/3.9.13/lib/python3.9/site-packages/pyarmor/protect_code2.pt
INFO     Start obfuscating the scripts...
INFO            /root/.phu/gen/foo.py -> dist/foo.py
INFO     Write patched script for debugging: /root/.phu/gen/foo.py.pyarmor-patched
INFO     Set PYARMOR_CC to "gcc"
DEBUG    Check license file "/root/.pyarmor/license.lic"
INFO     Generating spp code for <frozen foo> ...
DEBUG    build map from line no. to code object
DEBUG    Traverse syntax tree
DEBUG    Find 1 available functions
DEBUG    Got spp code for function "test1" at line 2
DEBUG    Find 1 top functions
DEBUG    Check code objects for super mode
DEBUG    Compile syntax tree to code object
DEBUG    Check code objects for spp mode
INFO     Generate module spp code successfully
INFO     Compiling spp code ...
DEBUG    CC: gcc
DEBUG    cflags: -O3 -c -fPIC -fno-stack-protector -fno-asynchronous-unwind-tables -fno-unwind-tables
INFO     Compile spp code successfully
Failed to reloc this entry, no symbol value
Traceback (most recent call last):
  File "/root/.pyenv/versions/3.9.13/bin/pyarmor", line 8, in <module>
    sys.exit(main_entry())
  File "/root/.pyenv/versions/3.9.13/lib/python3.9/site-packages/pyarmor/pyarmor.py", line 1608, in main_entry
    main(sys.argv[1:])
  File "/root/.pyenv/versions/3.9.13/lib/python3.9/site-packages/pyarmor/pyarmor.py", line 1600, in main
    args.func(args)
  File "/root/.pyenv/versions/3.9.13/lib/python3.9/site-packages/pyarmor/pyarmor.py", line 747, in _obfuscate
    encrypt_script(prokey, a, b, wrap_mode=args.wrap_mode,
  File "/root/.pyenv/versions/3.9.13/lib/python3.9/site-packages/pyarmor/utils.py", line 1060, in encrypt_script
    sppmode, co = build_co_module(lines, modname, encoding=encoding,
  File "/root/.pyenv/versions/3.9.13/lib/python3.9/site-packages/pyarmor/cobuilder.py", line 62, in build_co_module
    co = sppbuild(mtree, modname)
  File "/root/.pyenv/versions/3.9.13/lib/python3.9/site-packages/pyarmor/sppmode.py", line 63, in build_co
    co = fb((mtree, modname))
RuntimeError: Link object failed
```

Are there any way to solve this issue?

Thanks,

**Comment by jondy** [comment]
It may be a bug, I'll check this next month when I'm back to office.

**Comment by jondy** [comment]
Fixed in v7.7.0
